backend/test/v2/integration/cache_test.go

```diff
@@ -0,0 +1,296 @@
+
+	var newPipelineUploadClient func() (*apiServer.PipelineUploadClient, error)
+	var newPipelineClient func() (*apiServer.PipelineClient, error)
+	var newRunClient func() (*apiServer.RunClient, error)
+	var newRecurringRunClient func() (*apiServer.RecurringRunClient, error)
+
+	if *isKubeflowMode {
+		s.resourceNamespace = *resourceNamespace
+
+		newPipelineUploadClient = func() (*apiServer.PipelineUploadClient, error) {
+			return apiServer.NewKubeflowInClusterPipelineUploadClient(s.namespace, *isDebugMode)
+		}
+		newPipelineClient = func() (*apiServer.PipelineClient, error) {
+			return apiServer.NewKubeflowInClusterPipelineClient(s.namespace, *isDebugMode)
+		}
+		newRunClient = func() (*apiServer.RunClient, error) {
+			return apiServer.NewKubeflowInClusterRunClient(s.namespace, *isDebugMode)
+		}
+		newRecurringRunClient = func() (*apiServer.RecurringRunClient, error) {
+			return apiServer.NewKubeflowInClusterRecurringRunClient(s.namespace, *isDebugMode)
+		}
+	} else {
+		clientConfig := test.GetClientConfig(*namespace)
+
+		newPipelineUploadClient = func() (*apiServer.PipelineUploadClient, error) {
+			return apiServer.NewPipelineUploadClient(clientConfig, *isDebugMode)
+		}
+		newPipelineClient = func() (*apiServer.PipelineClient, error) {
+			return apiServer.NewPipelineClient(clientConfig, *isDebugMode)
+		}
+		newRunClient = func() (*apiServer.RunClient, error) {
+			return apiServer.NewRunClient(clientConfig, *isDebugMode)
+		}
+		newRecurringRunClient = func() (*apiServer.RecurringRunClient, error) {
+			return apiServer.NewRecurringRunClient(clientConfig, *isDebugMode)
+		}
+	}
+
+	var err error
+	s.pipelineUploadClient, err = newPipelineUploadClient()
+	if err != nil {
+		glog.Exitf("Failed to get pipeline upload client. Error: %s", err.Error())
+	}
+	s.pipelineClient, err = newPipelineClient()
+	if err != nil {
+		glog.Exitf("Failed to get pipeline client. Error: %s", err.Error())
+	}
+	s.runClient, err = newRunClient()
+	if err != nil {
+		glog.Exitf("Failed to get run client. Error: %s", err.Error())
+	}
+	s.recurringRunClient, err = newRecurringRunClient()
+	if err != nil {
+		glog.Exitf("Failed to get recurring run client. Error: %s", err.Error())
+	}
+
+	s.cleanUp()
+}
+
+func (s *CacheTestSuite) TestCacheRecurringRun() {
+	t := s.T()
+
+	pipelineVersion := s.preparePipeline()
+
+	createRecurringRunRequest := &recurringRunParams.RecurringRunServiceCreateRecurringRunParams{Body: &recurring_run_model.V2beta1RecurringRun{
+		DisplayName: "hello world",
+		Description: "this is hello world",
+		PipelineVersionReference: &recurring_run_model.V2beta1PipelineVersionReference{
+			PipelineID:        pipelineVersion.PipelineID,
+			PipelineVersionID: pipelineVersion.PipelineVersionID,
+		},
+		MaxConcurrency: 10,
+		Mode:           recurring_run_model.RecurringRunModeENABLE,
+		Trigger: &recurring_run_model.V2beta1Trigger{
+			PeriodicSchedule: &recurring_run_model.V2beta1PeriodicSchedule{
+				IntervalSecond: 60,
```

Comment:
Could we increase the interval to speed this up? Maybe it doesn't matter because the pipeline run may take around 60s.

Reply:
Yeah, that's why I set 60.